src/event.lib.php

```diff
@@ -727,17 +727,22 @@ public static function setNewPassword($user_id = 0)
             return false;
         }
 
+        /* invalidates the password for the given user */
+        $new_password = wp_generate_password(15, true, false);
+        wp_set_password($new_password, $user_id);
+
         $website = SucuriScan::getDomain();
         $user_login = $user->user_login;
         $display_name = $user->display_name;
-        $new_password = wp_generate_password(15, true, false);
+        $key = self::GetPasswordResetKey($user);
+        $reset_password_url = network_site_url("wp-login.php?action=rp&key=$key&login=" . rawurlencode($user_login), 'login' );
```

Comment:
What will/should happen here when `$key` is not valid?

Reply:
Thanks for catching this @gvre!
If we fail to generate the `$key` we display `Error` to the user. I updated this as part of https://github.com/Sucuri/sucuri-wordpress-plugin/pull/100/commits/b282f9d27a77e541822f131223a4ed83877bc3a5
<img width="1466" alt="error" src="https://user-images.githubusercontent.com/3615268/103660430-30778e00-4f33-11eb-9e48-8a64f99e72e5.png">

Reply:
Hm, as I can see [here](https://github.com/Sucuri/sucuri-wordpress-plugin/commit/b282f9d27a77e541822f131223a4ed83877bc3a5#diff-69f862d1bf875449c7883dea23c8ae299b248aa391b0764c128305ba3d717980R739), we return false when the `$key` is an error. Will the error be displayed in that case?

Reply:
Yes, it will: https://github.com/imgerson/sucuri-wordpress-plugin/blob/fix-pass-recovery/src/settings-posthack.php#L258-L266